test/ZoraNFTCreatorV1_Fork.t.sol

```diff
@@ -9,62 +10,157 @@ import {MockMetadataRenderer} from "./metadata/MockMetadataRenderer.sol";
+            creator = ZoraNFTCreatorV1(getDeployment().factory);
+            verifyAddressesFork(chainName);
+            forkEdition();
+            forkDrop();
+            forkDropGeneric();
+        }
+    }
+
+    function verifyAddressesFork(string memory chainName) internal {
+        ChainConfig memory chainConfig = getChainConfig();
+        DropDeployment memory deployment = getDeployment();
 
-            address factoryAddress = deployment.factory;
-            ZoraNFTCreatorV1 factory = ZoraNFTCreatorV1(factoryAddress);
+        assertEq(chainConfig.factoryOwner, OwnableUpgradeable(deployment.factory).owner(), string.concat("configured owner incorrect on: ", chainName));
 
-            assertEq(chainConfig.factoryOwner, OwnableUpgradeable(factoryAddress).owner(), string.concat("configured owner incorrect on: ", chainName));
-            assertEq(deployment.dropMetadata, address(factory.dropMetadataRenderer()), string.concat("configured drop metadata renderer incorrect on: ", chainName));
-            assertEq(deployment.editionMetadata, address(factory.editionMetadataRenderer()), string.concat("configured edition metadata renderer incorrect on: ", chainName));
-            // assertEq(deployment.dropImplementation, address(factory.editionMetadataRenderer()), string.concat("configured metadata renderer incorrect on: ", chainName));
+        bytes32 slot = UUPSUpgradeable(deployment.factoryImpl).proxiableUUID();
+        address factoryImpl = address(uint160(uint256(vm.load(deployment.factory, slot))));
+        if (factoryImpl != deployment.factoryImpl) {
 
-            address deployedEdition = factory.createEdition(
-                "name",
-                "symbol",
-                100,
-                500,
-                DEFAULT_FUNDS_RECIPIENT_ADDRESS,
-                DEFAULT_FUNDS_RECIPIENT_ADDRESS,
-                IERC721Drop.SalesConfiguration({
-                    publicSaleStart: 0,
-                    publicSaleEnd: type(uint64).max,
-                    presaleStart: 0,
-                    presaleEnd: 0,
-                    publicSalePrice: 0.1 ether,
-                    maxSalePurchasePerAddress: 0,
-                    presaleMerkleRoot: bytes32(0)
-                }),
-                "desc",
-                "animation",
-                "image"
-            );
-            ERC721Drop drop = ERC721Drop(payable(deployedEdition));
-            (, uint256 fee) = drop.zoraFeeForAmount(10);
-            vm.startPrank(DEFAULT_FUNDS_RECIPIENT_ADDRESS);
-            vm.deal(DEFAULT_FUNDS_RECIPIENT_ADDRESS, 10 ether + fee);
-            drop.purchase{value: 1 ether + fee}(10);
-            assertEq(drop.totalSupply(), 10);
-            
+            console2.log("===========");
+            console2.log("===========");
+            console2.log("FACTORY IMPL NOT SAME AS CHAIN: SIMULATING UPGRADE STEP");
+            console2.log("to save changes: call upgradeTo(", deployment.factoryImpl, ")");
```

Comment:
possible for this to spit out calldata or whatever input we would want to copy and paste to gnosis?

Reply:
👍